### PR TITLE
feat(GridForm)!: Simplified GridForm labels to allow React.ReactNode or be hidden

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- ~[ ] Related to Abstract designs:~
- [x] Related to JIRA ticket: GM-34
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Self-quoting my quote from the associated ticket:

> "We've been trying to get the GridForm to allow three kinds of labels:
> * **Visual**: the most common kind, where the label is shown as part of the form
> * **Hidden**: like visual, but not actually visible - it only gets narrated with the input
> * **External**: no label at all, but a unique id that's expected to be coincidentally the same as an external element
> 
> I think the problem becomes easier if we stop allowing for external labels, force users to provide a label always, and allow an optional hideLabel prop. That would allow for **visual** and **hidden** and be an easy to understand API."

You can see from the updated story that once the field is hidden, it just magically works:

Screenshot of an inline submit button aligned with a single text field

Note that this does not tackle resizing the submit button. That's a separate ask that I think Esi is looking at.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
@jakemhiller yes, I just need to get around to validating it works on the monolith. Hoping for.. next week on that 🤞

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.46-alpha.bbf672.0<br/>@codecademy/gamut@9.5.6-alpha.bbf672.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f15c9bb5c43d4d6cccdf3af--codecademy-styleguide.netlify.app>https://5f15c9bb5c43d4d6cccdf3af--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f15c9bb5c43d4d6cccdf3af>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.46-alpha.706204.0<br/>@codecademy/gamut@9.5.6-alpha.706204.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f15d5e7ac87bce4eece6af0--codecademy-styleguide.netlify.app>https://5f15d5e7ac87bce4eece6af0--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f15d5e7ac87bce4eece6af0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.46-alpha.cf7d15.0<br/>@codecademy/gamut@9.5.6-alpha.cf7d15.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f15e3d2f4da2082abfe56cc--codecademy-styleguide.netlify.app>https://5f15e3d2f4da2082abfe56cc--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f15e3d2f4da2082abfe56cc>Deploy Logs</a></p>

File: packages/styleguide/stories/Core/Organisms/GridForm.stories.mdx [hunk 1]
```diff
@@ -294,41 +294,18 @@ We can make the Submit button inline with an input by setting the column
 sizes so they fit on the same row (e.g size 8 for an input and size 4 for
 the submit).
 
-We can additionally remove the label from text inputs and checkbox inputs (though this
-causes alignment issues in checkboxes, as you can see below).
+We can additionally remove the label from text inputs and checkbox inputs.
+Use the `hideLabel` prop to remove the label, allowing the submit button to
```
Comment (afbreilyn) [inline packages/styleguide/stories/Core/Organisms/GridForm.stories.mdx:298]: <3

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@1.0.0-alpha.ef96f1.0<br/>@codecademy/gamut-labs@1.7.46-alpha.ef96f1.0<br/>@codecademy/gamut@10.0.0-alpha.ef96f1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f16fe089d8382d9430bb285--codecademy-styleguide.netlify.app>https://5f16fe089d8382d9430bb285--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f16fe089d8382d9430bb285>Deploy Logs</a></p>

**Pull request merged**